### PR TITLE
Improvements to reduce LDAR-Sim memory/resource usage

### PR DESCRIPTION
**Pull request by ThomasGalesloot**
Tested with end-to-end testing to ensure no functional changes to LDAR-Sim. Logs will be attached. 
[memory_improvement_logs.zip](https://github.com/LDAR-Sim/LDAR_Sim/files/11628920/memory_improvement_logs.zip)

### CODE REVIEW

**Review by KeeganShaw-GIS — APPROVED** [review]
Nice, makes sense to me

File: LDAR_Sim/src/initialization/sites.py
Comment (SEAJang) [inline LDAR_Sim/src/initialization/sites.py:1]: Could we update the docstring?

**Review by SEAJang — APPROVED** [review]
Other than updating an existing docstring, great job.

**Comment by ThomasGalesloot** [comment]
> 

Docstring is updated

**Comment by ThomasGalesloot** [comment]
> Other than updating an existing docstring, great job.

Docstring is now updated

**Pull request merged**